library/iFixit/Matryoshka/Scope.php

```diff
@@ -21,14 +21,20 @@ public function getPrefix() {
       return $this->scopePrefix ?: $this->getScopePrefix();
    }
 
-   public function getScopePrefix(bool $reset = false) {
+   public function getScopePrefix(bool $reset = false, bool $generateOnMiss = true) {
       if ($this->scopePrefix === null || $reset) {
-         $scopeValue = $this->backend->getAndSet($this->getScopeKey(),
-          function() {
-            return substr(md5(microtime() . $this->scopeName), 0, 16);
-         }, 0, $reset);
-
-         $this->scopePrefix = "{$scopeValue}-";
+         $scopeValue = $reset ? self::MISS : $this->backend->get($this->getScopeKey());
+         if ($scopeValue === self::MISS) {
+            if ($generateOnMiss) {
+               $scopeValue = substr(md5(microtime() . $this->scopeName), 0, 16);
+               $this->scopePrefix = "{$scopeValue}-";
+               $this->backend->set($this->getScopeKey(), $scopeValue);
+            } else {
+               return self::MISS;
+            }
+         } else {
+            $this->scopePrefix = "{$scopeValue}-";
+         }
```

Comment:
Minor: I think we could de-duplicate the `$this->scopePrefix = "{$scopeValue}-";` assignment here

```suggestion
         if ($scopeValue === self::MISS) {
            if ($generateOnMiss) {
               $scopeValue = substr(md5(microtime() . $this->scopeName), 0, 16);
               $this->backend->set($this->getScopeKey(), $scopeValue);
            } else {
               return self::MISS;
            }
         }
         $this->scopePrefix = "{$scopeValue}-";
```